components/improv_wifi/wifi_provisioning.c

```diff
@@ -190,73 +221,135 @@ bool improv_wifi_connect(const char *ssid, const char *password) {
   return true;
 }
 
-/**
- */
 bool improv_wifi_is_connected(void) {
-  bool ret = network_if_get_ip(NULL);
+  wifi_ap_record_t apRec;
+
+  if (esp_wifi_sta_get_ap_info(&apRec) == ESP_OK) {
+    return true;
+  }
 
-  return ret;
+  return false;
 }
 
-/**
- */
 void improv_wifi_get_local_ip(uint8_t *address) {
   esp_netif_ip_info_t ip_info;
 
-  if (network_if_get_ip(&ip_info) == false) {
-    ESP_LOGE(TAG, "%s: no valid IP available", __func__);
-
-    return;
+  /* Wait a short time for an IP to become available. Improv frequently asks
+     for the local IP during onboarding; when called too early we should
+     wait briefly rather than returning 0.0.0.0 immediately. */
+  const TickType_t timeout = pdMS_TO_TICKS(2000); /* 2s */
+  TickType_t start = xTaskGetTickCount();
+
+  while (wifi_get_ip(&ip_info) == false) {
+    if ((xTaskGetTickCount() - start) > timeout) {
+      ESP_LOGW(TAG, "%s: no valid IP available after timeout", __func__);
+      /* return all-zero address (caller should handle this) */
+      address[0] = 0;
+      address[1] = 0;
+      address[2] = 0;
+      address[3] = 0;
+      return;
+    }
+    vTaskDelay(pdMS_TO_TICKS(100));
   }
 
-  address[0] = ip_info.ip.addr >> 0;
-  address[1] = ip_info.ip.addr >> 8;
-  address[2] = ip_info.ip.addr >> 16;
-  address[3] = ip_info.ip.addr >> 24;
+  address[0] = (uint8_t)(ip_info.ip.addr & 0xFF);
+  address[1] = (uint8_t)((ip_info.ip.addr >> 8) & 0xFF);
+  address[2] = (uint8_t)((ip_info.ip.addr >> 16) & 0xFF);
+  address[3] = (uint8_t)((ip_info.ip.addr >> 24) & 0xFF);
 
   ESP_LOGI(TAG, "%d.%d.%d.%d", address[0], address[1], address[2], address[3]);
 }
 
-/**
- */
 void improv_init(void) {
   uint8_t webPortStr[6] = {0};
   uint16_t webPort = CONFIG_WEB_PORT;
   uint8_t urlStr[26] = "http://{LOCAL_IPV4}:";
 
-  ESP_LOGI(TAG, "start IPROV WiFi provisioning service");
-
   utoa(webPort, (char *)webPortStr, 10);
   strcat((char *)urlStr, (char *)webPortStr);
 
   improv_wifi_create();
   improv_wifi_serialWrite(uart_write);
-  improv_wifi_set_device_info(CF_ESP32, "esp32_snapclient", "0.0.3",
+
+  // Automatically detect chip family based on build target
+  uint8_t chipFamily = CF_ESP32;
+#if CONFIG_IDF_TARGET_ESP32S3
+  chipFamily = CF_ESP32_S3;
+#elif CONFIG_IDF_TARGET_ESP32S2
+  chipFamily = CF_ESP32_S2;
+#elif CONFIG_IDF_TARGET_ESP32C3
+  chipFamily = CF_ESP32_C3;
+#elif CONFIG_IDF_TARGET_ESP32
+  chipFamily = CF_ESP32;
+#endif
+
+  improv_wifi_set_device_info(chipFamily, "esp32_snapclient", "0.0.3",
```

Comment:
Do you have an idea how to make the version string available across different components and main, etc.? We don't need to do this right away but this should be done at some point. I'd like to match it github releases at some point too without too much manual work